### DISCUSSION TITLE
Should there be a "try-catch" in defined endpoint method block?

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#2372] [Q&A] [answered]
In the documentation demo file, I see there's no "try-catch" at all in method definition for class endpoint on server side like the following screenshot:

![image](https://github.com/serverpod/serverpod/assets/32797520/243e2d5a-de22-4b4e-8632-610d2554b18b)

My question is, is it necessary at all to add a "try-catch" on the server side? I believe it's a good practice to always add "try-catch" for future methods on flutter side though.

**Comment by SandPod** [comment, marked as answer]
Good question!

In tutorials and examples, it is often common to leave out the error-handling side of things. Usually because it adds unnecessary context that might confuse the reader.

In Serverpod,  If an unhandled exception is thrown in an endpoint the server will respond with a 500 response. This might be exactly what you want to communicate in some scenarios.

The client will then throw these 500 responses as a `ServerpodClientInternalServerError` exception (that is a subclass of ServerpodClientException).

So you are not forced to try-catch the db calls on the server side, but depending on your desired behavior you might want to implement them.

**Reply by gazialankus** [reply]
And the uncaught exceptions thrown in endpoints can be found in `serverpod_session_log` table. In case you may expect them to be in the `serverpod_log` table this explanation should be helpful.
